lib/lib.go

```diff
@@ -88,7 +88,7 @@ type InstanceOptions struct {
 	keyStore                key.Store
 	profiles                profile.Store
 	bus                     event.Bus
-	collectionSet           collection.Set
+	collectionSet           collection.WritableSet
```

Comment:
Lines like this make me especially think that `collection.Set` should just have methods for adding rather than having a separate `collection.WritableSet` interface.